app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -95,7 +161,64 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
         showToolbar(getString(R.string.library_downloads))
     }
 
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        val menuRes = when (downloadStore.state.mode) {
+            DownloadFragmentState.Mode.Normal -> R.menu.library_menu
+            is DownloadFragmentState.Mode.Editing -> R.menu.download_select_multi
+        }
+        inflater.inflate(menuRes, menu)
+
+        menu.findItem(R.id.delete_downloads_multi_select)?.title =
+            SpannableString(getString(R.string.bookmark_menu_delete_button)).apply {
+                setTextColor(requireContext(), R.attr.destructive)
+            }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean = when (item.itemId) {
+        R.id.close_history -> {
+            close()
+            true
+        }
+
+        R.id.delete_downloads_multi_select -> {
+            deleteDownloadItems(downloadStore.state.mode.selectedItems)
+            downloadStore.dispatch(DownloadFragmentAction.ExitEditMode)
+            true
+        }
+        R.id.open_downloads -> {
+            val selectedDownloads = downloadStore.state.mode.selectedItems
+            val listOfDownload = selectedDownloads.toList()
+            context?.let {
+                AbstractFetchDownloadService.openFile(
+                    context = it,
+                    contentType = listOfDownload[0].contentType,
+                    filePath = listOfDownload[0].filePath
+                )
+            }
+            true
+        }
+        else -> super.onOptionsItemSelected(item)
+    }
+
+    private fun getMultiSelectSnackBarMessage(downloadItems: Set<DownloadItem>): String {
+        return if (downloadItems.size > 1) {
+            getString(R.string.download_delete_multiple_items_snackbar)
+        } else {
+            String.format(
+                requireContext().getString(
+                    R.string.history_delete_single_item_snackbar
+                ), downloadItems.first().filePath.toShortUrl(requireComponents.publicSuffixList)
```

Comment:
I think it will be better to just show the file name as the path could get pretty long.
![image](https://user-images.githubusercontent.com/773158/96631755-bd239000-12e4-11eb-8f5e-cecbd998ddae.png)